emails/views.py

```diff
@@ -424,9 +418,34 @@ def _handle_reply(from_address, message_json):
             # TODO: send the user an email
             # that replies are a premium feature
             incr_if_enabled('free_user_reply_attempt', 1)
-            return HttpResponse(
-                "Rely replies require a premium account", status=403
+            return False
+    else:
+        # The From: is not a Relay user, so make sure this is a reply *TO* a
+        # premium Relay user
+        try:
+            [to_local_portion, to_domain_portion] = to_address.split('@')
+            address = _get_address(
+                to_address, to_local_portion, to_domain_portion
             )
+            user_profile = address.user.profile_set.first()
+            if not user_profile.has_premium:
+                incr_if_enabled('free_user_reply_attempt', 1)
+                return False
```

Comment:
Should be check the in reply to address being a premium relay user that the external is sending to and NOT the `to_address` which will have `replies` portion according to the check [here](https://github.com/mozilla/fx-private-relay/blob/main/emails/views.py#L300-L301).

If we want to disable external messages that are responses to the premium reply feature, we should be checking the in reply to while checking the local portion at inbound email checks [here](https://github.com/mozilla/fx-private-relay/blob/main/emails/views.py#L300-L301)